app/rdf/query-latest-cube-iri.ts

```diff
@@ -0,0 +1,43 @@
+/** Creates SPARQL query to fetch latest cube iri.
+ * Works for both versioned and unversioned cubes.
+ */
+export const getLatestCubeIriQuery = (cubeIri: string) => {
+  return `PREFIX cube: <https://cube.link/>
+PREFIX schema: <http://schema.org/>
+
+SELECT ?iri WHERE {
+  {
+    # Versioned cube.
+    SELECT ?iri WHERE {
+      VALUES ?oldIri { <${cubeIri}> }
+      ?versionHistory schema:hasPart ?oldIri .
+      ?versionHistory schema:hasPart ?iri .
+      ?iri schema:version ?version .
+      ?iri schema:creativeWorkStatus ?status .
+      ?oldIri schema:creativeWorkStatus ?oldStatus .
+      FILTER(NOT EXISTS { ?iri schema:expires ?expires . } && ?status IN (?oldStatus, <https://ld.admin.ch/vocabulary/CreativeWorkStatus/Published>))
+    }
+  } UNION {
+    {
+      # Non-versioned cube.
+      SELECT ?iri WHERE {
+        VALUES ?iri { <${cubeIri}> }
+        ?iri cube:observationConstraint ?shape .
+        ?iri schema:creativeWorkStatus ?status .
+        FILTER(NOT EXISTS { ?iri schema:expires ?expires . } && ?status = <https://ld.admin.ch/vocabulary/CreativeWorkStatus/Published>)      
+      }
+    }
+  } UNION {
+    {
+      # Version history of a cube.
+      VALUES ?versionHistory { <${cubeIri}> }
+      ?versionHistory schema:hasPart ?iri .
+      ?iri schema:version ?version .
+      ?iri schema:creativeWorkStatus ?status .
+      FILTER(NOT EXISTS { ?iri schema:expires ?expires . } && ?status = <https://ld.admin.ch/vocabulary/CreativeWorkStatus/Published>)
```

Comment:
this would redirect version history to published cubes exclusively. 
OTOH there might be cube histories with only draft cubes. 
what do you think might this something to ORDER BY DESC ?

Reply:
Again, I am not sure if we should default to showing draft datasets in Visualize PROD, even if they are newer than published versions. I think it makes perfect sense when it comes to versioned cubes, as we can exactly know its status, if there's a "generic" IRI, I would assume it should led to last published cube. Let me know if that's not the expectation 👍 

How would it work with ordering? I think we would need to somehow know the "position" of a given status, right?

Reply:
Based on the cubeHistory I'd show Draft only in cases no Published are available :+1: 
i.e. subquery
```sparql
SELECT ?iri {
    VALUES ?versionHistory { <${cubeIri}> }
      ?versionHistory schema:hasPart ?iri .
      ?iri schema:version ?version .
      ?iri schema:creativeWorkStatus ?status .
      FILTER(NOT EXISTS { ?iri schema:expires ?expires . } )
} 
ORDER BY DESC (?status) DESC(?version) # order by status then version (to prefer Published over latest)
LIMIT 1
```

Reply:
I hope the new query should correctly catch all the cases 🤞